node/src/crypto/asymmetric_key_ext.rs

```diff
@@ -93,13 +93,13 @@ impl AsymmetricKeyExt for SecretKey {
     fn generate_ed25519() -> Result<Self, Error> {
         let mut bytes = [0u8; Self::ED25519_LENGTH];
         getrandom::getrandom(&mut bytes[..]).expect("RNG failure!");
```

Comment:
Maybe we can extend the `Error` variant here to include these `rand::Error`s and our crypto errors?

Reply:
Extended the `Error` variant with a new one called `RNGFailure`